packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -149,19 +153,18 @@ contract TaikoL1 is
      * @param blockId The index of the block to prove. This is also used to
      *        select the right implementation version.
      * @param inputs A list of data input:
-     *        - inputs[0] An Evidence object with various information regarding
-     *          the block to be proven and the actual proofs.
-     *        - inputs[1] The target block to be proven invalid.
-     *        - inputs[2] The receipt for the `invalidBlock` transaction
-     *          on L2. Note that the `invalidBlock` transaction is supposed to
-     *          be the only transaction in the L2 block.
+     *        - inputs[0] The target block's metadata.
+     *        - inputs[1] The circuit ID which is submitted by the block
+     *          proposer when the block is proposed.
+     *        - inputs[2] The signature zk-proof which is submitted by the
```

Comment:
some not-in-depth thoughts: if we use a two-stage-challenge strategy for `proveBlockInvalid` since the proof verification is not cheap, can we use the same challenge strategy for `proveBlockValid`? Like there is a 5 mins challenge window for valid block proof? which may also make `proveBlockValid` a lot cheaper, since most of the provers are honest?

Reply:
Not sure I understand the challenge window part. But all proofs needs to be validated on-chain, and sooner is better. If the proof isn't' validated on-chain it's just an optimistic rollup with a validity proof as the fraud proof, so you don't get the benefits of a zk-rollup because the fraud proof delay needs to remain the same, only the downsides.

If I misunderstand and the idea is to batch prove multiple proofs that where first only the proof data is submitted on-chain and then another prover takes that data and proves that those proofs are valid then that would indeed be more efficient, at the cost of slower verified status on-chain. So you could see that batching working over block proofs, but you could see the proposer proofs as just any other block proof that could also get batch proven this way over multiple blocks, but in the end it's kind of the same I think. But this batching over multiple blocks is very likely the way to go in the future, proof generation time just needs to be fast enough.

Reply:
Yah right what I mean is separating the proof submission and proof validation, at the cost of slower block status validation👍